### PR TITLE
[4.0] keystone: Reorganize HA/fernet code

### PR DESCRIPTION
**Pull request by skazi0**
Moved most of the fernet setup to ha.rb. Some redundancy was added but
it is now easier to work with the (more-complex) HA case.

The main goal was to reorder step so that fernet keys are configured and
distributed to cluster nodes before apache is (re)started. This way, there
is no time when apache would be configured to use fernet keys but no actual
keys would be present. The problem was detected in cluster extension scenario
where new node was added to an already existing cluster.

### CODE REVIEW

File: chef/cookbooks/keystone/providers/fernet.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+property :rsync_command, String, default: ""
```
Comment (jsuchome) [inline chef/cookbooks/keystone/providers/fernet.rb]: Doesn't this go into a resource?

Comment (skazi0) [reply chef/cookbooks/keystone/providers/fernet.rb]: Yeah, that's a benefit of using software which is so old that the docs are already removed from the official site :o)

**Comment by jsuchome** [comment]
https://github.com/crowbar/crowbar-openstack/pull/1787 was merged, you can rebase

**Comment by skazi0** [comment]
rebased and added some more details in commit messages

**Pull request merged**